js/src/vercel.ts

```diff
@@ -797,83 +836,74 @@ export class AISDKExporter {
-            // - trace_id: use from the existing run tree
-            if (traceMap.interop.type === "traceable") {
-              override = {
-                id: override.id,
-                parent_run_id:
-                  override.parent_run_id ?? traceMap.interop.parentRunTree.id,
-                dotted_order: [
-                  traceMap.interop.parentRunTree.dotted_order,
-                  override.dotted_order,
-                ]
-                  .filter(Boolean)
-                  .join("."),
-                trace_id: traceMap.interop.parentRunTree.trace_id,
-              };
-            } else if (traceMap.interop.type === "user") {
-              // Allow user to specify custom trace ID = run ID of the root run
-              // - id: use user provided run ID if root run, otherwise preserve
-              // - parent_run_id: use user provided run ID if root run, otherwise preserve
-              // - dotted_order: replace the trace_id with the user provided run ID
-              // - trace_id: use user provided run ID
-              const userTraceId = traceMap.interop.userTraceId ?? uuid4();
-              override = {
-                id:
-                  override.id === override.trace_id ? userTraceId : override.id,
-                parent_run_id:
-                  override.parent_run_id === override.trace_id
-                    ? userTraceId
-                    : override.parent_run_id,
-                dotted_order: override.dotted_order.replace(
-                  override.trace_id,
-                  userTraceId
-                ),
-                trace_id: userTraceId,
-              };
+            if (task.item.interop?.type === "traceable") {
+              actions.push({
+                type: "reparent",
+                runId: task.item.id,
+                parentDotOrder: task.item.interop.parentRunTree.dotted_order,
+              });
             }
+          } else {
+            actions.push({ type: "delete", runId: task.item.id });
           }
 
-          sampled.push([override, task.item.run]);
           task.item.sent = true;
         }
 
         const children = traceMap.childMap[task.item.id] ?? [];
         queue.push(
-          ...children.map((child) => {
-            return {
-              item: child,
-              dottedOrder: [
-                task.dottedOrder,
-                convertToDottedOrderFormat(
-                  child.startTime,
-                  child.id,
-                  child.executionOrder
-                ),
-              ].join("."),
-              traceId: task.traceId,
-            };
-          })
+          ...children.map((item) => ({
+            item,
+            dotOrder: joinDotOrder(task.dotOrder, getDotOrder(item)),
+          }))
         );
       }
     }
 
-    this.logDebug(`sampled runs to be sent to LangSmith`, sampled);
+    let processed = sampled;
+
+    for (const action of actions) {
```

Comment:
Why do this in a separate loop instead of inline above?

Reply:
Think you're right, all of these actions are affecting the consequent nodes only, will inline